File: summer2013/openmp/c/45_first_touch.c
#include <stdio.h>

int main()
{
	int i;
	double *A;

	A = (double *) malloc(sizeof(double)*N);

#pragma omp parallel for
	for(i=0; i<N; i++)
		A[i] = 0.0;

#pragma omp parallel for
	for(i=0; i<N; i++) {
		// ......
	}
}

File: summer2013/openmp/fortran/45_first_touch.f90
program first_touch
    use omp_lib
    implicit none
    integer(8) :: i
    double precision :: A(N)

!$omp parallel do
    do i=1, N
        A(i) = 0.0
    end do
!$omp end parallel do

!$omp parallel do
    do i=1, N
        !......
    end do
!$omp end parallel do

end
